operator/v1/types_ingress.go

```diff
@@ -632,6 +632,30 @@ const (
 	NeverHTTPHeaderPolicy IngressControllerHTTPHeaderPolicy = "Never"
 )
 
+// IngressControllerHTTPUniqueIdHeaderPolicy describes configuration for a
+// unique id header.
+type IngressControllerHTTPUniqueIdHeaderPolicy struct {
+	// name specifies the name of the HTTP header (for example, "unique-id")
+	// that the IngressController should inject into HTTP requests.  The
+	// field's value must be a valid HTTP header name as defined in RFC 2616
+	// section 4.2.  This field has no effect unless both name and format
+	// are specified.
+	//
+	// +optional
+	// +kubebuilder:validation:Pattern="^$|^[-!#$%&'*+.0-9A-Z^_`a-z|~]+$"
+	Name string `json:"name,omitempty"`
+
+	// format specifies the format for the injected HTTP header's value.
+	// This field has no effect unless both name and format are specified.
+	// For the HAProxy-based IngressController implementation, this format
+	// uses the same syntax as the HTTP log format; see the corresponding
```

Comment:
no regex?

Reply:
Added.  It's a little loose because the syntax and grammar for format strings are a little loose and because HAProxy adds new formats and converters from time to time.  It does catch some errors though, such as missing delimiters or bogus control characters.